doc/manual_experimental.rst

```diff
@@ -1731,324 +1585,304 @@ e.g. with given example `echo("ab")` will be rewritten just once:
+  template optPeg{peg(pattern)}(pattern: string{lit}): Peg =
+    var gl {.global, gensym.} = peg(pattern)
+    gl
 
-Since version 1.4 a stricter definition of "side effect" is available. In addition
-to the existing rule that a side effect is calling a function with side effects
-the following rule is also enforced:
+  for i in 0 .. 3:
+    echo match("(a b c)", peg"'(' @ ')'")
+    echo match("W_HI_Le", peg"\y 'while'")
 
-Any mutation to an object does count as a side effect if that object is reachable
-via a parameter that is not declared as a `var` parameter.
+The `optPeg` template optimizes the case of a peg constructor with a string
+literal, so that the pattern will only be parsed once at program startup and
+stored in a global `gl` which is then re-used. This optimization is called
+hoisting because it is comparable to classical loop hoisting.
 
-For example:
+
+AST based overloading
+=====================
+
+Parameter constraints can also be used for ordinary routine parameters; these
+constraints affect ordinary overloading resolution then:
 
 .. code-block:: nim
 
-  {.experimental: "strictFuncs".}
+  proc optLit(a: string{lit|`const`}) =
+    echo "string literal"
+  proc optLit(a: string) =
+    echo "no string literal"
 
-  type
-    Node = ref object
-      le, ri: Node
-      data: string
+  const
+    constant = "abc"
 
-  func len(n: Node): int =
-    # valid: len does not have side effects
-    var it = n
-    while it != nil:
-      inc result
-      it = it.ri
+  var
+    variable = "xyz"
 
-  func mut(n: Node) =
-    let m = n # is the statement that connected the mutation to the parameter
-    m.data = "yeah" # the mutation is here
-    # Error: 'mut' can have side effects
-    # an object reachable from 'n' is potentially mutated
+  optLit("literal")
+  optLit(constant)
+  optLit(variable)
 
+However, the constraints `alias` and `noalias` are not available in
+ordinary routines.
 
-The algorithm behind this analysis is described in
-the `view types section <#view-types-algorithm>`_.
 
+Parallel & Spawn
+================
 
-View types
-==========
+Nim has two flavors of parallelism:
+1) `Structured`:idx: parallelism via the `parallel` statement.
+2) `Unstructured`:idx: parallelism via the standalone `spawn` statement.
 
-**Note**:  `--experimental:views`:option: is more effective
-with `--experimental:strictFuncs`:option:.
+Nim has a builtin thread pool that can be used for CPU intensive tasks. For
+IO intensive tasks the `async` and `await` features should be
+used instead. Both parallel and spawn need the `threadpool <threadpool.html>`_
+module to work.
 
-A view type is a type that is or contains one of the following types:
+Somewhat confusingly, `spawn` is also used in the `parallel` statement
+with slightly different semantics. `spawn` always takes a call expression of
+the form `f(a, ...)`. Let `T` be `f`'s return type. If `T` is `void`
+then `spawn`'s return type is also `void` otherwise it is `FlowVar[T]`.
```

Comment:
```suggestion
the form `f(a, ...)`. Let `T` be `f`'s return type. If `T` is `void`,
then `spawn`'s return type is also `void`, otherwise it is `FlowVar[T]`.
```